src/platform/error_abi.cpp

```diff
@@ -0,0 +1,191 @@
+            }
+            else
+            {
+                *object = nullptr;
+                return xlang_hresult_no_interface;
+            }
+            AddRef();
+            return 0;
+        }
+
+        uint32_t XLANG_CALL AddRef() noexcept final
+        {
+            return ++m_count;
+        }
+
+        uint32_t XLANG_CALL Release() noexcept final
+        {
+            auto result = --m_count;
+            if (result == 0)
+            {
+                delete this;
+            }
+            return result;
+        }
+
+        void GetError(xlang_result* error) noexcept override
+        {
+            *error = m_result;
+        }
+
+        void GetMessage(xlang_string* message) noexcept override
+        {
+            *message = m_message;
+        }
+
+        void GetLanguageError(xlang_string* language_error) noexcept override
+        {
+            *language_error = m_language_error;
+        }
+
+        void GetExecutionTrace(xlang_unknown** execution_trace) noexcept override
+        {
+            m_execution_trace.copy_to(execution_trace);
+        }
+
+        void GetProjectionIdentifier(xlang_string* projection_identifier) noexcept override
+        {
+            *projection_identifier = m_projection_identifier;
+        }
+
+        void GetLanguageInformation(xlang_unknown** language_information) noexcept override
+        {
+            m_language_information.copy_to(language_information);
+        }
+
+        void GetPropagatedError(xlang_error_info** propagated_error) noexcept override
+        {
+            m_next_propagated_error.copy_to(propagated_error);
+        }
+
+        void PropagateError(
+            xlang_string projection_identifier,
+            xlang_string language_error,
+            xlang_unknown* execution_trace,
+            xlang_unknown* language_information
+        ) noexcept override
+        {
+            // If the error info can't be modified, don't collect infomration on propagations
+            // as this error info can be reused for another error.
+            if (!m_modifiable)
+            {
+                return;
+            }
+
+            com_ptr<xlang_error_info> propagated_error;
+            propagated_error.attach(
+                xlang_originate_error(
+                    m_result,
+                    m_message,
+                    projection_identifier,
+                    language_error,
+                    execution_trace,
+                    language_information));
+
+            error_info* last_propagated_error = this;
+            while (last_propagated_error->m_next_propagated_error != nullptr)
+            {
+                last_propagated_error = static_cast<error_info*>(last_propagated_error->m_next_propagated_error.get());
+            }
+            last_propagated_error->m_next_propagated_error = propagated_error;
+        }
+
+    private:
+        xlang_result m_result{};
+        xlang_string m_message{ nullptr };
```

Comment:
Are these handle types? If so, where is the destructor that releases them?

Reply:
Good catch, I will have a PR to address this and your other comment.